src/audio/pipeline/pipeline-stream.c

```diff
@@ -257,9 +257,54 @@ int pipeline_trigger(struct pipeline *p, struct comp_dev *host, int cmd)
 		/* Execute immediately */
 		ret = pipeline_trigger_run(p, host, cmd);
 		return ret == PPL_STATUS_PATH_STOP ? 0 : ret;
+	case COMP_TRIGGER_XRUN:
+		/*
+		 * XRUN can happen on a pipeline, not directly attached to the host,
+		 * find the original one
+		 */
+		do {
```

Comment:
This looks like it should be a separate function ?

Reply:
sorry, missed this one, let me do that